runtime/gc_vlhgc/AllocationContextBalanced.cpp

```diff
@@ -545,7 +545,7 @@ MM_AllocationContextBalanced::recycleRegion(MM_EnvironmentVLHGC *env, MM_HeapReg
 void
 MM_AllocationContextBalanced::tearDownRegion(MM_EnvironmentBase *env, MM_HeapRegionDescriptorVLHGC *region)
 {
-	MM_MemoryPoolBumpPointer *memoryPool = (MM_MemoryPoolBumpPointer*)region->getMemoryPool();
+	MM_MemoryPoolAddressOrderedList *memoryPool = (MM_MemoryPoolAddressOrderedList *)region->getMemoryPool();
```

Comment:
tearDown is virtual, this might work without explicit casting

Reply:
tearDown in class MemoryPool is protected method, we need to move it to public for working without explicit casting.

Reply:
ok, I don't understand why it would be protected, but let's not change or debate it now (in general, I understand valid reasons to do it, but not in this particular case). 
so, let's just keep the explicit cast.